inlanecruising_plugin/src/inlanecruising_plugin.cpp

```diff
@@ -616,8 +667,50 @@ std::vector<PointSpeedPair> InLaneCruisingPlugin::maneuvers_to_points(const std:
       points_and_target_speeds.push_back(pair);
     }
   }
+    // Here we are limiting the trajectory length to the given length by maneuver end dist as opposed to the end of lanelets involved.
+    double starting_route_downtrack = wm_->routeTrackPos(points_and_target_speeds.front().point).downtrack; 
+    size_t i = 0;
+    size_t max_i = 0;
+    double dist_accumulator = starting_route_downtrack;
+    lanelet::BasicPoint2d prev_point;
+
+    double ending_downtrack = maneuvers.back().lane_following_maneuver.end_dist;
+    if(ending_downtrack + config_.buffer_ending_downtrack < wm_->getRouteEndTrackPos().downtrack){
```

Comment:
If ending_downtrack + buffer is > route end, does it make more sense to just set end downtrack to route end? Right now if our buffer is 20m and the ending downtrack is 19m, we still don't use any buffer at all.

Reply:
good point. I added another clause